### PR TITLE
Implement Ably Kinesis Rule Resource

### PR DESCRIPTION
**Pull request by graham-russell**
### Implement Ably Rule Resource with Kinesis Target
As a team, we decided to implement the Ably rule with a Kinesis target first. This PR includes the following items
- tfplugindocs generated documentation
- Ably Kinesis rule resource with inbuilt source and target. 
- Updates to provider and models to represent Ably Rule for Kinesis target
- Playground tf templates updated
- Resource example updated for kinesis integration rule
- Acceptance testing configured

### CODE REVIEW

File: examples/playground/rule_aws_kinesis.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+resource "ably_rule_kinesis" "rule0" {
+  app_id    = ably_app.app0.id
+  status    = "enabled"
+  rule_type = "aws/kinesis"
```
Comment (Morganamilo) [inline examples/playground/rule_aws_kinesis.tf:4]: This shouldn't be needed any more

File: examples/playground/rule_aws_kinesis.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  aws_authentication = {
+    mode              = "credentials",
+    access_key_id     = "hhhh"
+    secret_access_key = "ffff"
+  }
+  # aws_authentication = {
+  #     mode = "assumeRole",
+  #     role_arn = "cccc"
+  # }
+  kinesis_target = {
```
Comment (Morganamilo) [inline examples/playground/rule_aws_kinesis.tf:18]: This should just be target.

File: internal/provider/resource_ably_rule_kinesis.go [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:        types.StringType,
+				Computed:    true,
+				Description: "The rule ID.",
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					tfsdk_resource.UseStateForUnknown(),
+				},
+			},
+			"app_id": {
```
Comment (Morganamilo) [inline internal/provider/resource_ably_rule_kinesis.go:29]: Needs RequiresReplace

File: internal/provider/resource_ably_rule_kinesis.go [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+					"role_arn": {
+						Type:     types.StringType,
+						Optional: true,
+					},
+					"access_key_id": {
+						Type:      types.StringType,
+						Optional:  true,
+						Sensitive: true,
+					},
+					"secret_access_key": {
```
Comment (Morganamilo) [inline internal/provider/resource_ably_rule_kinesis.go:78]: Can you really not edit the key without replacing it?

File: internal/provider/resource_ably_rule_kinesis.go [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+		return
+	}
+
+	// Remove resource from state
+	resp.State.RemoveResource(ctx)
+}
+
+// Import resource
+func (r resourceRule) ImportState(ctx context.Context, req tfsdk_resource.ImportStateRequest, resp *tfsdk_resource.ImportStateResponse) {
+	tfsdk_resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
```
Comment (Morganamilo) [inline internal/provider/resource_ably_rule_kinesis.go:432]: Should this be appid+id like the other resources?

**Review by Morganamilo — CHANGES_REQUESTED** [review]

File: examples/playground/rule_aws_kinesis.tf (hunk 1 above)
Comment (graham-russell) [reply examples/playground/rule_aws_kinesis.tf]: Fixed

File: examples/playground/rule_aws_kinesis.tf (hunk 2 above)
Comment (graham-russell) [reply examples/playground/rule_aws_kinesis.tf]: Modified

File: internal/provider/resource_ably_rule_kinesis.go (hunk 4 above)
Comment (graham-russell) [reply internal/provider/resource_ably_rule_kinesis.go:78]: You're right. It's not required so I've removed it.

File: internal/provider/resource_ably_rule_kinesis.go (hunk 5 above)
Comment (graham-russell) [reply internal/provider/resource_ably_rule_kinesis.go]: Modified

**Review by Morganamilo — APPROVED** [review]

**Comment by graham-russell** [comment]
NOTE: There's an issue with the staging control API endpoint that we use for testing. Because of this, tests are currently failing. I've used the prod endpoint to verify that acceptance tests work correctly. Output: 

`~/Documents/src/terraform-provider-ably (feature/IAC-44-Ably-Rule-Resource ✗) make testacc
TF_ACC=1 go test $(go list ./... | grep -v 'vendor') -v  -timeout 120m
?       terraform-provider-ably [no test files]
=== RUN   TestAccAblyApp
--- PASS: TestAccAblyApp (10.72s)
=== RUN   TestAccAblyKey
--- PASS: TestAccAblyKey (11.47s)
=== RUN   TestAccAblyNamespace
--- PASS: TestAccAblyNamespace (11.87s)
=== RUN   TestAccAblyQueue
--- PASS: TestAccAblyQueue (12.19s)
=== RUN   TestAccAblyRuleKinesis
--- PASS: TestAccAblyRuleKinesis (9.73s)
PASS`

**Pull request merged**